deps/rabbit/src/rabbit_classic_queue.erl

```diff
@@ -509,15 +555,18 @@ recover_durable_queues(QueuesAndRecoveryTerms) ->
 capabilities() ->
     #{unsupported_policies => [%% Stream policies
                                <<"max-age">>, <<"stream-max-segment-size-bytes">>,
-                               <<"queue-leader-locator">>, <<"initial-cluster-size">>,
+                               <<"initial-cluster-size">>,
                                %% Quorum policies
                                <<"delivery-limit">>, <<"dead-letter-strategy">>, <<"max-in-memory-length">>, <<"max-in-memory-bytes">>, <<"target-group-size">>],
       queue_arguments => [<<"x-expires">>, <<"x-message-ttl">>, <<"x-dead-letter-exchange">>,
                           <<"x-dead-letter-routing-key">>, <<"x-max-length">>,
                           <<"x-max-length-bytes">>, <<"x-max-priority">>,
                           <<"x-overflow">>, <<"x-queue-mode">>, <<"x-queue-version">>,
                           <<"x-single-active-consumer">>, <<"x-queue-type">>,
-                          <<"x-queue-master-locator">>],
+                          <<"x-queue-leader-locator">>] ++ case rabbit_queue_location:master_locator_permitted() of
```

Comment:
When
```
deprecated_features.permit.queue_master_locator = false
```
is set, this line leads to the following error being logged by `rabbit_queue_type` module:
```
2024-07-08 09:46:56.396527+02:00 [error] <0.1265.0> Deprecated features: `queue_master_locator`: Feature `queue_master_locator` is deprecated.
2024-07-08 09:46:56.396527+02:00 [error] <0.1265.0> Its use is not permitted per the configuration (overriding the default, which is permitted):
2024-07-08 09:46:56.396527+02:00 [error] <0.1265.0>     "deprecated_features.permit.queue_master_locator = false"
2024-07-08 09:46:56.396527+02:00 [error] <0.1265.0> Its use will not be permitted by default in a future minor RabbitMQ version and the feature will be removed from a future major RabbitMQ version; actual versions to be determined.
2024-07-08 09:46:56.396527+02:00 [error] <0.1265.0> To continue using this feature when it is not permitted by default, set the following parameter in your configuration:
2024-07-08 09:46:56.396527+02:00 [error] <0.1265.0>     "deprecated_features.permit.queue_master_locator = true"
```
which is poor user experience as there is actually no error.
`rabbit_deprecated_features:is_permitted/1` assumes only to be called if the old feature is actually used. However, here, this function is also called when the old feature is not used.